lib/email/receiver.rb

```diff
@@ -208,8 +208,13 @@ def process_internal
 
       if user.present?
         log_and_validate_user(user)
-      else
-        raise UserNotFoundError unless SiteSetting.enable_staged_users
+      elsif !SiteSetting.enable_staged_users
+        category = Category.find_by_email(mail.to)
+        if category&.email_in_allow_strangers == true
+          user = Discourse.system_user
+        else
+          raise UserNotFoundError
```

Comment:
Is it okay that we raise "user not found" when category is not found? Or could it be confusing later when debugging/logging? 🤔

Reply:
Good question. Technically in this block of code we are looking for user - author of post. This is exception that if we try to send email to `Category.mail_to` we create staged user or if staged users option is disabled we are using system user. 

It means that if category does not exists, user was not found.